### ISSUE TITLE
Registering machines again using auth-key

### ISSUE THREAD

**Issue by jbajic** [tailscale/tailscale#4865]
### What is the issue?

The issues arise only if the host has been registered on a tailscale, then removed, and then you try to add it again, the `tailscale up --auth-key <auth-key> --advertise-tags=tag:<tag>` hangs. 

It is possible to add it without using auth-key flag

### Steps to reproduce

1. Install tailscale on Mac using this method => https://github.com/tailscale/tailscale/wiki/Tailscaled-on-macOS
2. Run the tailscale command for adding the machine into a VPN `tailscale up --auth-key <auth-key> --advertise-tags=tag:<tag>`
3. Remove the machine from the VPN in tailscale settings.
4. Change the hostname of the machine.
5. Run tailscale logout
6. Try to add the machine again by running the same command
7. The command hangs indefinitely 

### OS

macOS

### OS version

Mac OS 12.1,  Darwin strange.local 21.2.0 , ARM

### Tailscale version

1.27.0-dev

### Bug report

BUG-45317af73ed0a96aa03e868289a884eef4441ada62c95e282addb676c028ec6a-20220616094428Z-653726a83ecb7ecf

**Comment by DentonGentry** [comment]
Was the authkey a Reusable authkey, or one-time use? If one-time then I think this is expected: even though it is the same machine, it is the second authentication. The key only works for the first one.

**Comment by jbajic** [comment]
> Was the authkey a Reusable authkey, or one-time use? If one-time then I think this is expected: even though it is the same machine, it is the second authentication. The key only works for the first one.

Yes the key was reusable, I reproduced this issue twice, on different Mac machines

**Comment by jikaczmarski** [comment]
Just want to jump in to keep this alive. I'm experiencing very similar behavior with Tailscale.
